wagtail/admin/forms/models.py

```diff
@@ -142,14 +143,20 @@ def defer_required_fields(self):
             except KeyError:
                 pass
 
-        for formset in self.formsets.values():
+        for name, formset in self.formsets.items():
             for form in formset:
                 form.defer_required_fields()
+            if formset.min_num is not None:
+                self.deferred_formset_min_nums[name] = formset.min_num
```

Comment:
Do we need to check for `and name not in self.deferred_formset_min_nums` as well, in case `defer_required_fields` has been called?

Without it, if `defer_required_on_fields` is empty, and the form has formsets with `min_num`, calling `defer_required_fields` multiple times might result in `deferred_formset_min_nums[name]` being `0` instead of the original value.

Reply:
Looks like it still hasn't been picked up, looking into it...

Reply:
Also, just want to note that Django's formsets behaviour in regards to `min_num` is now working against us: it always populates the form with at least `min_num` number of forms.

So if you have a non-deferrable required field in a formset with `min_num` e.g. 3, and you save as draft with fewer than 3 forms, the save is successful – but it gives you a form with 3 items in the formset, so you won't be able to save again without deleting the remaining empty ones added by Django.